src/frontend/src/optimizer/rule/trivial_project_to_values_rule.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::{BoxedRule, Rule};
+use crate::optimizer::plan_node::LogicalValues;
+use crate::optimizer::PlanRef;
+
+pub struct TrivialProjectToValuesRule {}
+impl Rule for TrivialProjectToValuesRule {
+    fn apply(&self, plan: PlanRef) -> Option<PlanRef> {
+        let project = plan.as_logical_project()?;
```

Comment:
maybe we can only do the optimization when the logical project's input is a values operator or other operators which has the determined output row count

Reply:
We need to know the exact row count. For instance, for a project on simple agg, will the result of simple agg be `None`?

Reply:
Let's just stick to values for now :cry:

Reply:
I think I will use a row counter.